examples/Azure_IoT_Central_ESP32_AzureIoTKit/AzureIoT.cpp

```diff
@@ -949,15 +957,18 @@ static int get_mqtt_client_config_for_dps(azure_iot_t* azure_iot, mqtt_client_co
   password_span = split_az_span(data_buffer_span, MQTT_PASSWORD_BUFFER_SIZE, &data_buffer_span);
   EXIT_IF_TRUE(is_az_span_empty(password_span), RESULT_ERROR, "Failed reserving buffer for password_span.");
 
-  password_length = generate_sas_token_for_dps(
-    &azure_iot->dps_client,
-    azure_iot->config->device_key,
-    azure_iot->config->sas_token_lifetime_in_minutes,
-    data_buffer_span,
-    azure_iot->config->data_manipulation_functions,
-    password_span,
-    &azure_iot->sas_token_expiration_time);
-  EXIT_IF_TRUE(password_length == 0, RESULT_ERROR, "Failed creating mqtt password for DPS connection.");    
+  if(!is_az_span_empty(azure_iot->config->device_key))
```

Comment:
Do the replacement suggested above somewhere, do not use is_az_span_empty, use az_span_is_content_equal against AZ_SPAN_EMPTY. Here and everywhere if possible.